src/elements/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+import * as React from "react";
+import clsx from "clsx";
+import Box, { BoxProps } from "@vertigis/web/ui/Box";
+import Checkbox from "@vertigis/web/ui/Checkbox";
+import ChevronDownIcon from "@vertigis/web/ui/icons/ChevronDown";
+import Collapse from "@vertigis/web/ui/Collapse";
+import DynamicIcon, { DynamicIconProps } from "@vertigis/web/ui/DynamicIcon";
+import { FormElementProps, FormElementRegistration } from "@geocortex/workflow/runtime";
+import IconButton, { IconButtonProps } from "@vertigis/web/ui/IconButton";
+import Link from "@vertigis/web/ui/Link";
+import List, { ListProps } from "@vertigis/web/ui/List";
+import ListItemIcon from "@vertigis/web/ui/ListItemIcon";
+import ListItemText from "@vertigis/web/ui/ListItemText";
+import ListItemButton from "@vertigis/web/ui/ListItemButton";
+import { UIContext } from "@vertigis/web/ui/UIContext";
+import { BrandingService } from "@vertigis/web/branding/BrandingService";
+/**
+ * Properties for the `Tree` component.
+ */
+export declare type TreeViewProps = {
+    showCheckBoxes?: boolean;
+};
+
+const IconButtonStyleOverrides: IconButtonProps["sx"] = {
+    // Hack required to override .gcx-forms.defaults
+    borderRadius: "50% !important",
+    // Hack required to match disabled list item content
+    "&.Mui-disabled": {
+        opacity: 0.38,
+    },
+};
+
+type SettableBoxProps = Pick<BoxProps, "maxHeight" | "maxWidth">;
+
+type SettableTreeProps = Pick<ListProps, "dense" | "subheader">;
+
+type SettableCheckBoxProps = Pick<TreeViewProps, "showCheckBoxes">;
+
+interface TreeElementNode {
+    key: string;
```

Comment:
What's the purpose of this? We are only using it as the React key. Usually the component would pick the key from other available information (or generate one), rather than accept it as a prop.

Reply:
Removed